Assignments for lesson 03

Swt16 opened the pull request:
Assignments for lesson 03.

natasha-aleksandrova commented on students/ShinTran/lesson03/SlicingLab.py:
```diff
@@ -0,0 +1,52 @@
+	if type(sequence) is tuple:
+		return (last,) + mid + (first,)
+	else:
+		return last + mid + first
```
couple comments here:
* per pep8: function names and var names should be snake_case -> `exchange_first_last`
* your indentation needs to be exactly 4 spaces (no tabs)
* this slicing assignment should be purely slicing, not indexing. [start:stop:step] is slicing / [index] is indexing
* there should be no special logic for the type. same code should work for string, tuple or list (the case when using slicing)

natasha-aleksandrova commented on students/ShinTran/lesson03/SlicingLab.py:
```diff
@@ -0,0 +1,52 @@
+	if type(sequence) is tuple and (len(firstThird) == 1):
+		return midLast + firstThird
+	else:
+		return midLast + firstThird
```
should be no special logic here 😉

natasha-aleksandrova commented on students/ShinTran/lesson03/list_lab.py:
```diff
@@ -0,0 +1,88 @@
+print(series1)
+for i in range(0,len(series1)):
+	if (series1[i][0] == "P"):
+		print(series1[i])
```
your loop should be:
```
for fruit in fruits:
    if fruit[0] == "P":
        print(fruit)
```

natasha-aleksandrova commented on students/ShinTran/lesson03/mailroom.py:
```diff
@@ -0,0 +1,106 @@
+	d10 = ["Mary Johnson", 12689.07]
+	d11 = ["Michael David", 57145.50]
+	d12 = ["Patricia Brown", 81679.15]
+	masterList = [d1,d2,d3,d4,d5,d6,d7,d8,d9,d10,d11,d12]
```
why use anonymous vars?
```
donors = [
    ["James Smith", 342334],
    [Mary Johsnon", 3029843]
]
```
is more clear and pythonic

natasha-aleksandrova commented on students/ShinTran/lesson03/mailroom.py:
```diff
@@ -0,0 +1,106 @@
+	createNameList()
+	if userInput == 'list':
+		for i in range(0,len(nameList)):
+			print(nameList[i])
```
please review looping over lists; this pattern is in many places. looping over list:
```
for name in name_list:
    print(name)
```

natasha-aleksandrova commented:
Per instructions on the website please post Pull Request link in the submission box. 
For this particular Pull Request the link is:

`https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/813`

Simply copy the link from the browser bar after you create the Pull Request.

Swt16 commented:
I just reuploaded the mailroom assignment. I fixed the variables so it uses snake case format. I also updated the indentation from tabs to 4 spaces. For loops was also updated. I initially coded it in a certain way where it was failing for me, so I ended up using the for in range instead. I think I'm just too used to Java/C# style coding. Either way, Python is something that I wish to improve on!

natasha-aleksandrova commented:
Understandable! If you use Pycharm (which is free and awesome) it comes with all the tools out of the box to encourage good python coding style 🙂
